src/components/wizards/create-vm/CreateVMWizard.js

```diff
@@ -0,0 +1,366 @@
+import React from 'react';
+import { Form, FormControl, Col, ButtonGroup, DropdownButton, MenuItem, Checkbox, FormGroup, ControlLabel, HelpBlock, Wizard } from 'patternfly-react';
+import './CreateVMWizard.css';
+import * as _ from 'lodash';
+import { StorageTable } from './StorageTable';
+
+const defaultState = {
+  activeStepIndex: 0,
+  basicSettings: {},
+  wizardValid: false
+}
+
+export class CreateVMWizard extends React.Component {
+
+    constructor(props){
+      super(props);
+      this.state = defaultState;
+    }
+  
+    onFormChange = (newValue, target) => {
+      let valid = null;
+      if (this.basicSettings[target].required && newValue.trim().length === 0){
+        valid= `${this.basicSettings[target].title} is required`;
+      }
+      const basicSettings = {
+        ...this.state.basicSettings,
+        [target]: {
+          value: newValue,
+          valid: valid
+        }
+      };
+      this.setState({
+        basicSettings: basicSettings
+      });
+      this.validateWizard(basicSettings);
+    }
+  
+    validateWizard = values => {
+      let wizardValid = true;
+
+      //check if all required fields are defined
+      const requiredKeys = Object.keys(this.basicSettings).filter(key => this.isFieldRequired(key, values));
+      const requiredKeysInValues = Object.keys(values).filter(key => this.isFieldRequired(key, values));
+      if(requiredKeys.length !== requiredKeysInValues.length){
+        wizardValid = false;
+      }
+
+      //check if all fields are valid
+      for (let key in values) {
+        if (values[key].valid && (this.basicSettings[key].isVisible ? this.basicSettings[key].isVisible(values) : true)) {
+          wizardValid = false;
+          break;
+        }
+      }
+
+      this.setState({
+        wizardValid: wizardValid
+      });
+    }
+
+    isFieldRequired = (key, basicSettings) => {
+      return this.basicSettings[key].required && (this.basicSettings[key].isVisible ? this.basicSettings[key].isVisible(basicSettings) : true);
+    }
+  
+    basicSettings = {
```

Comment:
What's the relation between component instance `this.basicSettings` and `this.state.basicSettings` ? (Do we need the second one?)

Reply:
Also, this looks like a declaration of form fields to be *interpreted* by the wizard component.

This is a good idea, as it separates "what" from "how" in terms of form fields.

In future, I'd like us to extract this concept into smaller, independent and reusable bits, including the common validation logic.